ui/src/components/Button.tsx

```diff
@@ -0,0 +1,35 @@
+import clsx from 'clsx'
+import { JSXElement, Show } from 'solid-js'
+
+/**
+ * Renders a button component with an optional loading state.
+ * If the loading state is passed the button will be disabled and show
+ * a loader ball while loading.
+ *
+ */
+export function Button(props: {
+  children: JSXElement[] | JSXElement
+  onClick?: () => void
+  isLoading?: boolean
+  type?: 'submit' | 'button' | 'reset'
+  variant?: 'primary' | 'secondary' | 'error' | 'success'
```

Comment:
maybe nice to make them compliant with daisyui docs? https://daisyui.com/components/button/

'btn-primary' 'btn-secondary' 'btn-neutral' 'btn-error', warning, info, etc are all about colors.

'btn-outline', 'btn-dash' 'btn-ghost' etc are all about style

'btn-xs', 'btn-sm', 'btn-md', 'btn-lg' and 'btn-xl' are all about size.

Maybe we can add those three with these options?

Reply:
Yes but i wanted to do that in a seperate pr, where i will rewrite all "ghost" and "small" buttons to this component as wel.

Reply:
Ah okay. Well in that case mergy mergy